datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -986,6 +986,22 @@ macro_rules! general_repeat_list {
     }};
 }
 
+/// Array_empty SQL function
+pub fn array_empty(args: &[ArrayRef]) -> Result<ArrayRef> {
+    let array = as_list_array(&args[0])?;
+    let builder = array
+        .iter()
+        .map(|arr| {
+            if let Some(arr) = arr {
+                Some(arr.is_empty())
+            } else {
+                Some(true)
```

Comment:
Should `NULL` rather map to `NULL`? Because if you want to default `NULL` to something, you can always wrap the method with `COALESCE`.

Reply:
> Should `NULL` rather map to `NULL`? Because if you want to default `NULL` to something, you can always wrap the method with `COALESCE`.

Since we intend to have the same behavior as Clickhouse does for this function, we can just return `true` for `null`.
Btw, not sure what `COALESCE` you mean.

Reply:
> Hi @crepererum , I do not fully understand your suggestion.

So your PR proposes that `array_empty(NULL)` is `TRUE` while I would argue that `array_empty(NULL)` should yield `NULL` because this is how most SQL functions work.

> Since we intend to have the same behavior as Clickhouse does for this function, we can just return `true` for `null`.

I would argue that DataFusion should not necessarily copy weird / unexpected behaviors of other vendors. @alamb what's your opinion here?

> Btw, not sure what `COALESCE` you mean.

This is a widely supported SQL method for "replace NULL w/ a default value", see [postgres docs](https://www.postgresql.org/docs/current/functions-conditional.html#FUNCTIONS-COALESCE-NVL-IFNULL).

Reply:
I would prefer return null if input is null, since it is different from an empty array.

Reply:
It make senses, I will support `NULL`

Reply:
Looks like this was resolved in [8e5f079](https://github.com/apache/arrow-datafusion/pull/7313/commits/8e5f079ba0148e159dfa50fad817e19365d4bf61)